### PR TITLE
Add a toast message and auto scroll to the 1st error field when failed to create/update a monitor or destination

### PR DESCRIPTION
**Pull request by ftianli-amzn**
*Issue #, if available:*

*Description of changes:*

### Improve the UX when facing form validation errors in creating/updating a monitor or destination:
 - Add a toast error message if the form validation fails when user attempts to create or edit a monitor.
The toast will dismiss in 10 seconds unless user click X button.
 - Auto scroll the page to the first field with error, and focus the field.

(^ click to see big image )

image

### Improve the UX when facing REST API errors in creating/updating a monitor:
- Add a toast error message to show the REST API errors when user attempts to create or edit a monitor.
The toast will dismiss in 20 seconds unless user click X button.
- Auto scroll the page to the top.

image

### Unit test changes:
- Add the path for `ui/notify` module in `moduleNameMapper` (https://github.com/opendistro-for-elasticsearch/index-management-kibana-plugin/blob/master/test/jest.config.js#L24)
- Mock the `ui/notify` module in jest set-up file to test `toastNotifications` without actually hitting the API (https://github.com/opendistro-for-elasticsearch/index-management-kibana-plugin/blob/master/test/setup.jest.ts#L45)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js [hunk 1]
```diff
@@ -136,7 +137,7 @@ export default class CreateMonitor extends Component {
           initialValues={initialValues}
           onSubmit={this.onSubmit}
           validateOnChange={false}
-          render={({ values, errors, handleSubmit, isSubmitting }) => (
+          render={(props) => (
```
Comment (ftianli-amzn) [inline public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js:140]: I'm not sure if this change is better.

**Comment by ftianli-amzn** [comment]
yarn test:jest
```
Test Suites: 6 skipped, 78 passed, 78 of 84 total
Tests:       20 skipped, 366 passed, 386 total
Snapshots:   127 passed, 127 total
Time:        31.982s
Ran all test suites.
✨  Done in 33.48s.

**Comment by ftianli-amzn** [comment]
```
Test Suites: 6 skipped, 80 passed, 80 of 86 total
Tests:       20 skipped, 370 passed, 390 total
Snapshots:   131 passed, 131 total
Time:        43.862s
Ran all test suites.
✨  Done in 45.26s.

File: public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js [hunk 2]
```diff
@@ -127,6 +131,15 @@ export default class CreateMonitor extends Component {
     else this.onCreate(monitor, formikBag);
   }
 
+  backendErrorHandler(actionName, data) {
+    toastNotifications.addDanger({
+      title: `Failed to ${actionName} the monitor`,
+      text: data.resp,
+      toastLifeTimeMs: 30000, // the default is 10 sec
```
Comment (skkosuri-amzn) [inline public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js:138]: Could you please double check this? During testing, I remember seeing this message forever, till I clicked on [X]

Comment (ftianli-amzn) [reply public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js:138]: Sure, I will double check it tomorrow. The toasts for backend errors are expected to appear for 30 seconds, and we can adjust the time if needed.

Comment (ftianli-amzn) [reply public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js:138]: Hi, after checking, I can see the toast message dismisses after 30 sec, and should we shorten the duration?

Comment (skkosuri-amzn) [reply public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js:138]: Thanks for checking. 30secs seems long, lets change it to 20sec

Comment (ftianli-amzn) [reply public/pages/CreateMonitor/containers/CreateMonitor/CreateMonitor.js:138]: Got it.

**Review by skkosuri-amzn — APPROVED** [review]

**Review by seraphjiang — APPROVED** [review]
LGTM, Thanks for the improvement.

**Pull request merged**